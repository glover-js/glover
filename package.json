{
  "name": "root",
  "private": true,
  "devDependencies": {
    "@babel/core": "^7.2.0",
    "@babel/preset-env": "^7.2.0",
    "@babel/preset-flow": "^7.0.0",
    "babel-core": "^7.0.0-bridge.0",
    "babel-eslint": "^10.0.1",
    "babel-jest": "^23.6.0",
    "coveralls": "^3.0.2",
    "eslint": "^5.10.0",
    "eslint-config-airbnb": "^17.1.0",
    "eslint-config-prettier": "^3.3.0",
    "eslint-plugin-flowtype": "^3.2.0",
    "eslint-plugin-import": "^2.14.0",
    "eslint-plugin-jsx-a11y": "^6.1.2",
    "eslint-plugin-markdown": "^1.0.0-rc.0",
    "eslint-plugin-prettier": "^3.0.0",
    "eslint-plugin-react": "^7.11.1",
    "flow-bin": "^0.87.0",
    "jest": "^23.6.0",
    "lerna": "^3.6.0",
    "prettier": "^1.15.3",
    "prettier-eslint": "^8.8.2",
    "rollup": "^0.67.4",
    "rollup-plugin-babel": "^4.0.3",
    "rollup-plugin-commonjs": "^9.2.0"
  },
  "scripts": {
    "build": "rollup -c",
    "build-clean": "rm -rf ./packages/*/dist",
    "clean-all": "rm -rf ./node_modules && rm -rf ./packages/*/node_modules && yarn build-clean",
    "dev": "rollup -c -w",
    "test": "jest",
    "test:all": "yarn typecheck && yarn lint && yarn test",
    "test:ci": "yarn typecheck && yarn lint && yarn coverage && yarn coveralls",
    "typecheck": "flow check --include-warnings",
    "lint": "eslint . --ext js,md",
    "coverage": "jest --coverage",
    "coveralls": "cat ./coverage/lcov.info | coveralls"
  },
  "files": [
    "dist"
  ],
  "jest": {
    "collectCoverageFrom": [
      "<rootDir>/packages/**/src/**/*.js"
    ],
    "coverageDirectory": "<rootDir>/coverage"
  }
}
