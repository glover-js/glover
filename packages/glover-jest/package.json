{
  "name": "glover-jest",
  "description": "TBD",
  "version": "1.0.0",
  "main": "dist/index.cjs.js",
  "module": "dist/index.esm.js",
  "dependencies": {
    "jest-matcher-utils": "^23.6.0"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/glover-js/glover"
  },
  "bugs": {
    "url": "https://github.com/glover-js/glover/issues"
  },
  "homepage": "https://glover-js.github.io/",
  "license": "MIT",
  "keywords": [
    "TBD"
  ]
}
