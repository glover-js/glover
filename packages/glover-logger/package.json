{
  "name": "glover-logger",
  "description": "TBD",
  "version": "1.0.0",
  "main": "dist/index.cjs.js",
  "module": "dist/index.esm.js",
  "repository": {
    "type": "git",
    "url": "https://github.com/glover-js/glover"
  },
  "bugs": {
    "url": "https://github.com/glover-js/glover/issues"
  },
  "homepage": "https://glover-js.github.io/",
  "license": "MIT",
  "keywords": [
    "TBD"
  ],
  "dependencies": {
    "chalk": "^2.4.1"
  }
}
