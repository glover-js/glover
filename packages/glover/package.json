{
  "name": "glover",
  "description": "TBD",
  "version": "1.0.0",
  "main": "dist/index.cjs.js",
  "module": "dist/index.esm.js",
  "bin": {
    "glover": "dist/bin/cli.js"
  },
  "dependencies": {
    "glover-jest": "1.0.0",
    "glover-logger": "1.0.0",
    "glover-resolver": "1.0.0",
    "inquirer": "^6.2.1",
    "mustache": "^3.0.1",
    "ora": "^3.0.0",
    "yargs": "^12.0.5"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/glover-js/glover"
  },
  "bugs": {
    "url": "https://github.com/glover-js/glover/issues"
  },
  "homepage": "https://glover-js.github.io/",
  "license": "MIT",
  "keywords": [
    "TBD"
  ]
}
