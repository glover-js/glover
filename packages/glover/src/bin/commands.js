// @flow

export const brew = {
  cmd: ['brew [recipe]', '$0'],
  desc: 'Brew a recipe',
  builder: (yargs: any) => {
    yargs.positional('recipe', {
      type: 'string',
      describe: 'The recipe you want to brew',
    });
    yargs.option('skipInstall', {
      type: 'boolean',
      describe: 'Skip npm/yarn install',
      default: false,
    });
    yargs.option('skipIngredient', {
      type: 'array',
      describe: 'Skip ingredients in the recipe',
      default: [],
    });
    yargs.option('skipStep', {
      type: 'array',
      describe: 'Skip steps in the recipe',
      default: [],
    });
  },
};

export const pick = {
  cmd: 'pick [ingredients...]',
  desc: 'Pick ingredients from a recipe',
  builder: (yargs: any) => {
    yargs.positional('ingredients', {
      type: 'array',
      describe: 'The ingredients you want to pick',
    });
    yargs.option('prompt', {
      alias: 'p',
      type: 'boolean',
      describe: '',
      default: false,
    });
    yargs.option('output', {
      alias: 'o',
      type: 'string',
      describe: 'Output path',
      default: process.cwd(),
    });
  },
};

export const compose = {
  cmd: 'compose [name] [ingredients...]',
  desc: 'compose a new recipe',
  builder: (yargs: any) => {
    yargs.positional('name', {
      type: 'string',
      describe: 'The name of your recipe',
    });
    yargs.positional('ingredients', {
      type: 'array',
      describe: 'Ingredients you want to include in your recipe',
    });
    yargs.option('extend', {
      alias: 'e',
      type: 'array',
      describe: 'Recipe(s) you want to extend',
      default: [],
    });
    yargs.option('output', {
      alias: 'o',
      type: 'string',
      describe: 'Output path',
      default: process.cwd(),
    });
  },
};
