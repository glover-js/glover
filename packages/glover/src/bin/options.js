// @flow
export const dryRun = {
  key: 'dryRun',
  options: {
    type: 'boolean',
    describe: 'Run the command without generating any files',
    default: false,
  },
};

export const silent = {
  key: 'silent',
  options: {
    type: 'boolean',
    describe: 'Prevent the command from printing anything to the console',
    default: false,
  },
};

export const noPrompt = {
  key: 'noPrompt',
  options: {
    type: 'boolean',
    describe: 'Prevent the command from prompting for values',
    default: false,
  },
};

export const debug = {
  key: 'debug',
  options: {
    type: 'boolean',
    describe: 'Print debugging info',
    default: false,
  },
};
