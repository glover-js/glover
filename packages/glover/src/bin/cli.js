// @flow
import cli from 'yargs';
import { brew, pick, compose } from './commands';
import { dryRun, silent, noPrompt, debug } from './options';
import run from '..';

export function parseArgv(argv: Array<string>): Promise<any> {
  return new Promise((resolve: any, reject: any) => {
    cli
      .scriptName('glover')
      .command(brew.cmd, brew.desc, brew.builder)
      .command(pick.cmd, pick.desc, pick.builder)
      .command(compose.cmd, compose.desc, compose.builder)
      .option(dryRun.key, dryRun.options)
      .option(silent.key, silent.options)
      .option(noPrompt.key, noPrompt.options)
      .option(debug.key, debug.options)
      .wrap(120)
      .parse(argv.slice(2), (err: any, parsedArgs: any, output: any) => {
        if (err) {
          return reject(err);
        }
        return resolve({
          arguments: parsedArgs,
          command: parsedArgs._[0] || 'brew',
          output,
          argv,
        });
      });
  });
}

if (process.env.NODE_ENV !== 'test') {
  parseArgv(process.argv)
    .then(args => {
      if (args.output) {
        console.log(args.output); // eslint-disable-line no-console
        process.exit(0);
      }
      return run(args);
    })
    .catch(err => {
      console.log(err); // eslint-disable-line no-console
      process.exit(1);
    });
}
