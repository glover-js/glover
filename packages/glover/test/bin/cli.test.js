import { parseArgv } from '../../src/bin/cli';
import { version } from '../../package.json';

describe('Tests for cli', () => {
  describe('command: brew', () => {
    it('defaults to brew', async () => {
      const { command } = await parseArgv(['./', './', 'recipe-test']);
      expect(command).toEqual('brew');
    });
    it('can be invoked with brew', async () => {
      const { command } = await parseArgv(['./', './', 'brew', 'recipe-test']);
      expect(command).toEqual('brew');
    });
    it('handles arguments to the recipe', async () => {
      const args = await parseArgv([
        './',
        './',
        'brew',
        'recipe-test',
        '--title',
        'example',
        '--license',
        'MIT',
      ]);
      expect(args.arguments.title).toEqual('example');
      expect(args.arguments.license).toEqual('MIT');
    });
    it('handles the --skipInstall flag', async () => {
      const argsA = await parseArgv([
        './',
        './',
        'recipe-test',
        '--skipInstall',
      ]);
      const argsB = await parseArgv(['./', './', 'recipe-test']);
      expect(argsA.arguments.skipInstall).toEqual(true);
      expect(argsB.arguments.skipInstall).toEqual(false);
    });
    it('handles the --skipIngredient flag', async () => {
      const argsA = await parseArgv([
        './',
        './',
        'recipe-test',
        '--skipIngredient',
        'a',
      ]);
      const argsB = await parseArgv([
        './',
        './',
        'recipe-test',
        '--skipIngredient',
        'a',
        'b',
        'c',
      ]);
      const argsC = await parseArgv(['./', './', 'recipe-test']);
      expect(argsA.arguments.skipIngredient).toEqual(['a']);
      expect(argsB.arguments.skipIngredient).toEqual(['a', 'b', 'c']);
      expect(argsC.arguments.skipIngredient).toEqual([]);
    });
    it('handles the --skipStep flag', async () => {
      const argsA = await parseArgv([
        './',
        './',
        'recipe-test',
        '--skipStep',
        'a',
      ]);
      const argsB = await parseArgv([
        './',
        './',
        'recipe-test',
        '--skipStep',
        'a',
        'b',
        'c',
      ]);
      const argsC = await parseArgv(['./', './', 'recipe-test']);
      expect(argsA.arguments.skipStep).toEqual(['a']);
      expect(argsB.arguments.skipStep).toEqual(['a', 'b', 'c']);
      expect(argsC.arguments.skipStep).toEqual([]);
    });
  });
  describe('command: pick', () => {
    it('handles the ingredients argument', async () => {
      const argsA = await parseArgv(['./', './', 'pick', 'index.html']);
      const argsB = await parseArgv(['./', './', 'pick', 'a', 'b']);
      expect(argsA.command).toEqual('pick');
      expect(argsA.arguments.ingredients).toEqual(['index.html']);
      expect(argsB.arguments.ingredients).toEqual(['a', 'b']);
    });
    it('handles additional arguments to the picked ingredient', async () => {
      const args = await parseArgv(['./', './', 'pick', 'a', '--title', 'b']);
      expect(args.arguments.ingredients).toEqual(['a']);
      expect(args.arguments.title).toEqual('b');
    });
    it('handles the --prompt flag', async () => {
      const argsA = await parseArgv(['./', './', 'pick', 'a', '--prompt']);
      const argsB = await parseArgv(['./', './', 'pick', 'a', '--p']);
      const argsC = await parseArgv(['./', './', 'pick', 'a']);
      expect(argsA.arguments.prompt).toEqual(true);
      expect(argsB.arguments.prompt).toEqual(true);
      expect(argsC.arguments.prompt).toEqual(false);
    });
    it('handles the --output flag', async () => {
      const argsA = await parseArgv(['./', './', 'pick', 'a', '--output', 'p']);
      const argsB = await parseArgv(['./', './', 'pick', 'a', '--o', 'p']);
      const argsC = await parseArgv(['./', './', 'pick', 'a']);
      expect(argsA.arguments.output).toEqual('p');
      expect(argsB.arguments.output).toEqual('p');
      expect(argsC.arguments.output).toEqual(process.cwd());
    });
  });
  describe('command: compose', () => {
    it('handles the name argument', async () => {
      const args = await parseArgv(['./', './', 'compose', 'recipe-name']);
      expect(args.command).toEqual('compose');
      expect(args.arguments.name).toEqual('recipe-name');
    });
    it('handles the ingredients argument', async () => {
      const argsA = await parseArgv(['./', './', 'compose', 'name', 'a']);
      const argsB = await parseArgv(['./', './', 'compose', 'name', 'a', 'b']);
      expect(argsA.arguments.ingredients).toEqual(['a']);
      expect(argsB.arguments.ingredients).toEqual(['a', 'b']);
    });
    it('handles the --extend flag', async () => {
      const argsA = await parseArgv([
        './',
        './',
        'compose',
        'name',
        '--extend',
        'a',
      ]);
      const argsB = await parseArgv([
        './',
        './',
        'compose',
        'name',
        '--e',
        'a',
        'b',
      ]);
      const argsC = await parseArgv(['./', './', 'compose', 'name']);
      expect(argsA.arguments.extend).toEqual(['a']);
      expect(argsB.arguments.extend).toEqual(['a', 'b']);
      expect(argsC.arguments.extend).toEqual([]);
    });
    it('handles the --output flag', async () => {
      const argsA = await parseArgv([
        './',
        './',
        'compose',
        'name',
        '--output',
        'p',
      ]);
      const argsB = await parseArgv(['./', './', 'compose', 'name', '-o', 'p']);
      const argsC = await parseArgv(['./', './', 'compose', 'name']);
      expect(argsA.arguments.output).toEqual('p');
      expect(argsB.arguments.output).toEqual('p');
      expect(argsC.arguments.output).toEqual(process.cwd());
    });
  });
  describe('global options', () => {
    it('handles the --dryRun flag', async () => {
      const argsA = await parseArgv(['./', './', 'recipe-test', '--dryRun']);
      const argsB = await parseArgv(['./', './', 'pick', 'a', '--dryRun']);
      const argsC = await parseArgv(['./', './', 'compose', 'b', '--dryRun']);
      expect(argsA.arguments.dryRun).toEqual(true);
      expect(argsB.arguments.dryRun).toEqual(true);
      expect(argsC.arguments.dryRun).toEqual(true);
    });
    it('handles the --silent flag', async () => {
      const argsA = await parseArgv(['./', './', 'recipe-test', '--silent']);
      const argsB = await parseArgv(['./', './', 'pick', 'a', '--silent']);
      const argsC = await parseArgv(['./', './', 'compose', 'b', '--silent']);
      expect(argsA.arguments.silent).toEqual(true);
      expect(argsB.arguments.silent).toEqual(true);
      expect(argsC.arguments.silent).toEqual(true);
    });
    it('handles the --noPrompt flag', async () => {
      const argsA = await parseArgv(['./', './', 'recipe-test', '--noPrompt']);
      const argsB = await parseArgv(['./', './', 'pick', 'a', '--noPrompt']);
      const argsC = await parseArgv(['./', './', 'compose', 'b', '--noPrompt']);
      expect(argsA.arguments.noPrompt).toEqual(true);
      expect(argsB.arguments.noPrompt).toEqual(true);
      expect(argsC.arguments.noPrompt).toEqual(true);
    });
    it('handles the --debug flag', async () => {
      const argsA = await parseArgv(['./', './', 'recipe-test', '--debug']);
      const argsB = await parseArgv(['./', './', 'pick', 'a', '--debug']);
      const argsC = await parseArgv(['./', './', 'compose', 'b', '--debug']);
      expect(argsA.arguments.debug).toEqual(true);
      expect(argsB.arguments.debug).toEqual(true);
      expect(argsC.arguments.debug).toEqual(true);
    });
  });
  describe('--help', () => {
    it('generates usage information', async () => {
      const { output } = await parseArgv(['./', './', '--help']);
      expect(output).toMatchSnapshot();
    });
    it('generates usage information for brew command', async () => {
      const { output } = await parseArgv(['path', 'path', 'brew', '--help']);
      expect(output).toMatchSnapshot();
    });
    it('generates usage information for pick command', async () => {
      const { output } = await parseArgv(['path', 'path', 'pick', '--help']);
      expect(output).toMatchSnapshot();
    });
    it('generates usage information for compose command', async () => {
      const { output } = await parseArgv(['path', 'path', 'compose', '--help']);
      expect(output).toMatchSnapshot();
    });
  });
  describe('--version', () => {
    it('generates version information', async () => {
      const { output } = await parseArgv(['path', 'path', '--version']);
      expect(output).toEqual(version);
    });
  });
});
