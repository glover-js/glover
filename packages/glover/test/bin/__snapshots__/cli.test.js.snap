// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for cli --help generates usage information 1`] = `
"glover [recipe]

Brew a recipe

Commands:
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe
  glover brew [recipe]                    Brew a recipe                                                        [default]
  glover pick [ingredients...]            Pick ingredients from a recipe
  glover compose [name] [ingredients...]  compose a new recipe

Positionals:
  recipe  The recipe you want to brew                                                                           [string]

Options:
  --help            Show help                                                                                  [boolean]
  --version         Show version number                                                                        [boolean]
  --dryRun          Run the command without generating any files                              [boolean] [default: false]
  --silent          Prevent the command from printing anything to the console                 [boolean] [default: false]
  --noPrompt        Prevent the command from prompting for values                             [boolean] [default: false]
  --debug           Print debugging info                                                      [boolean] [default: false]
  --skipInstall     Skip npm/yarn install                                                     [boolean] [default: false]
  --skipIngredient  Skip ingredients in the recipe                                                 [array] [default: []]
  --skipStep        Skip steps in the recipe                                                       [array] [default: []]"
`;

exports[`Tests for cli --help generates usage information for brew command 1`] = `
"glover brew [recipe]

Brew a recipe

Positionals:
  recipe  The recipe you want to brew                                                                           [string]

Options:
  --help            Show help                                                                                  [boolean]
  --version         Show version number                                                                        [boolean]
  --dryRun          Run the command without generating any files                              [boolean] [default: false]
  --silent          Prevent the command from printing anything to the console                 [boolean] [default: false]
  --noPrompt        Prevent the command from prompting for values                             [boolean] [default: false]
  --debug           Print debugging info                                                      [boolean] [default: false]
  --skipInstall     Skip npm/yarn install                                                     [boolean] [default: false]
  --skipIngredient  Skip ingredients in the recipe                                                 [array] [default: []]
  --skipStep        Skip steps in the recipe                                                       [array] [default: []]"
`;

exports[`Tests for cli --help generates usage information for compose command 1`] = `
"glover compose [name] [ingredients...]

compose a new recipe

Positionals:
  recipe       The recipe you want to brew                                                                      [string]
  name         The name of your recipe                                                                          [string]
  ingredients  Ingredients you want to include in your recipe                                      [array] [default: []]

Options:
  --help            Show help                                                                                  [boolean]
  --version         Show version number                                                                        [boolean]
  --dryRun          Run the command without generating any files                              [boolean] [default: false]
  --silent          Prevent the command from printing anything to the console                 [boolean] [default: false]
  --noPrompt        Prevent the command from prompting for values                             [boolean] [default: false]
  --debug           Print debugging info                                                      [boolean] [default: false]
  --skipInstall     Skip npm/yarn install                                                     [boolean] [default: false]
  --skipIngredient  Skip ingredients in the recipe                                                 [array] [default: []]
  --skipStep        Skip steps in the recipe                                                       [array] [default: []]
  --extend, -e      Recipe(s) you want to extend                                                   [array] [default: []]
  --output, -o      Output path                             [string] [default: \\"/Users/ollelauribostrom/Desktop/glover\\"]"
`;

exports[`Tests for cli --help generates usage information for pick command 1`] = `
"glover pick [ingredients...]

Pick ingredients from a recipe

Positionals:
  recipe       The recipe you want to brew                                                                      [string]
  ingredients  The ingredients you want to pick                                                    [array] [default: []]

Options:
  --help            Show help                                                                                  [boolean]
  --version         Show version number                                                                        [boolean]
  --dryRun          Run the command without generating any files                              [boolean] [default: false]
  --silent          Prevent the command from printing anything to the console                 [boolean] [default: false]
  --noPrompt        Prevent the command from prompting for values                             [boolean] [default: false]
  --debug           Print debugging info                                                      [boolean] [default: false]
  --skipInstall     Skip npm/yarn install                                                     [boolean] [default: false]
  --skipIngredient  Skip ingredients in the recipe                                                 [array] [default: []]
  --skipStep        Skip steps in the recipe                                                       [array] [default: []]
  --prompt, -p                                                                                [boolean] [default: false]
  --output, -o      Output path                             [string] [default: \\"/Users/ollelauribostrom/Desktop/glover\\"]"
`;
